src/internal/context/single-tab-stop-navigation-context.tsx

```diff
@@ -37,8 +38,13 @@ export function useSingleTabStopNavigation(focusable: null | FocusableDefinition
 
   useEffect(() => {
     if (focusable) {
-      const changeHandler = (element: null | Element, suppressed: boolean) =>
-        setFocusTargetActive(focusable.current === element || suppressed);
+      const changeHandler = (element: null | Element, suppressed: boolean) => {
```

Comment:
Is this just to avoid calls to `setFocusTargetActive`? Is this extra complexity worth it?

Reply:
Yes, the Lighthouse tests show a noticeable improvement when having this optimisation in place. However, I am changing it further as there is still a small performance impact, see https://github.com/cloudscape-design/components/pull/1901